src/Auth/AwsIamAuth.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace GraphQL\Auth;
+
+use Aws\Credentials\CredentialProvider;
+use GuzzleHttp\Psr7\Request;
+
+class AwsIamAuth implements AuthInterface
+{
+    protected const SERVICE_NAME = 'appsync';
+
+    /**
+     * @param Request $request
+     * @param array $headers
+     * @return Request
+     */
+    public function run(Request $request, array $headers = []): Request
+    {
+        $region = getenv('AWS_REGION');
+        $signature = new \Aws\Signature\SignatureV4('appsync', $region);
```

Comment:
If you composer "recommend" instead of "require" this package, I would check if this class exists before attempting to to instantiate and throw an exception notifying the user to install the package if they want to utilize this feature.